zebra-network/src/protocol/external/codec/tests.rs

```diff
@@ -0,0 +1,377 @@
+                .expect("message should be serialized");
+        }
+        bytes
+    });
+
+    rt.block_on(async {
+        let mut fr = FramedRead::new(Cursor::new(&v_bytes), Codec::builder().finish());
+        fr.next()
+            .await
+            .expect("a next message should be available")
+            .expect_err("that message should not deserialize")
+    });
+}
+
+#[test]
+fn max_msg_size_round_trip() {
+    use zebra_chain::serialization::ZcashDeserializeInto;
+
+    //let (rt, _init_guard) = zebra_test::init_async();
+    let _init_guard = zebra_test::init();
+
+    // make tests with a Tx message
+    let tx: Transaction = zebra_test::vectors::DUMMY_TX1
+        .zcash_deserialize_into()
+        .unwrap();
+    let msg = Message::Tx(tx.into());
+
+    use tokio_util::codec::{FramedRead, FramedWrite};
+
+    // i know the above msg has a body of 85 bytes
+    let size = 85;
+
+    // reducing the max size to body size - 1
+    zebra_test::MULTI_THREADED_RUNTIME.block_on(async {
+        let mut bytes = Vec::new();
+        {
+            let mut fw = FramedWrite::new(
+                &mut bytes,
+                Codec::builder().with_max_body_len(size - 1).finish(),
+            );
+            fw.send(msg.clone())
+                .await
+                .expect_err("message should not encode as it is bigger than the max allowed value");
+        }
+    });
+
+    // send again with the msg body size as max size
+    let msg_bytes = zebra_test::MULTI_THREADED_RUNTIME.block_on(async {
+        let mut bytes = Vec::new();
+        {
+            let mut fw = FramedWrite::new(
+                &mut bytes,
+                Codec::builder().with_max_body_len(size).finish(),
+            );
+            fw.send(msg.clone())
+                .await
+                .expect("message should encode with the msg body size as max allowed value");
+        }
+        bytes
+    });
+
+    // receive with a reduced max size
+    zebra_test::MULTI_THREADED_RUNTIME.block_on(async {
+        let mut fr = FramedRead::new(
+            Cursor::new(&msg_bytes),
+            Codec::builder().with_max_body_len(size - 1).finish(),
+        );
+        fr.next()
+            .await
+            .expect("a next message should be available")
+            .expect_err("message should not decode as it is bigger than the max allowed value")
+    });
+
+    // receive again with the tx size as max size
+    zebra_test::MULTI_THREADED_RUNTIME.block_on(async {
+        let mut fr = FramedRead::new(
+            Cursor::new(&msg_bytes),
+            Codec::builder().with_max_body_len(size).finish(),
+        );
+        fr.next()
+            .await
+            .expect("a next message should be available")
+            .expect("message should decode with the msg body size as max allowed value")
+    });
+}
+
+/// Check that the version test vector deserializes correctly without the relay byte
+#[test]
+fn version_message_omitted_relay() {
```

Comment:
this is the only new code in this file, the rest was moved here from `codec.rs`.